src/networking.c

```diff
@@ -4520,12 +4520,30 @@ void flushReplicasOutputBuffers(void) {
     }
 }
 
-mstime_t getPausedActionTimeout(uint32_t action) {
+char *getPausedPurpose(pause_purpose purpose) {
+    switch (purpose) {
+    case PAUSE_BY_CLIENT_COMMAND:
+        return "client_command";
+    case PAUSE_DURING_SHUTDOWN:
+        return "during_shutdown";
+    case PAUSE_DURING_FAILOVER:
+        return "during_failover";
```

Comment:
I like the INFO field, but I think the name and values are not perfect. There's something about the English language that doesn't sound right to me.

```
paused_purpose:client_command
paused_purpose:during_shutdown
paused_purpose:during_failover
```

I think the word "purpose" is not correct. Purpose is not what caused something to happen. The purpose is the end goal of some action, not the source.

To describe the source of the pause, I think "cause" or "reason" are better words. "paused_cause" or "paused_reason"?

And "during" is about time. It is the answer to when, not why. Maybe we can find a better word or just skip "during"?

How about "paused_reason:shutdown"? Or "paused_reason:shutdown_in_progress" or something like that?

Reply:
yean, paused_reason seems better, updated.